src/Microsoft.ML.AutoML/Experiment/Experiment.cs

```diff
@@ -196,6 +196,22 @@ public IList<TRunDetail> Execute()
                         "was reached, and the running MLContext was stopped. Details: {0}", e.Message);
                     return iterationResults;
                 }
+                catch (AggregateException e)
+                {
+                    // This exception is thrown when the IHost/MLContext of the trainer is canceled due to
+                    // reaching maximum experiment time. Simply catch this exception and return finished
+                    // iteration results. For some trainers, Like FastTree, because training is done in parallel
+                    // in can throw multiple OperationCancelledExceptions. This causes them to be returned as an
+                    // AggregateException and misses the first catch block. This is to handle that case.
+                    if (e.InnerExceptions.All( exception => exception is OperationCanceledException))
+                    {
+                        _logger.Warning("OperationCanceledException has been caught after maximum experiment time" +
+                        "was reached, and the running MLContext was stopped. Details: {0}", e.Message);
+                        return iterationResults;
+                    }
+
+                    Console.WriteLine(e.Message);
```

Comment:
I don't think we should be writing to the console in a library. Instead I think we should just

```suggestion
                    throw;
```

Reply:
Good catch. That was left over from my debugging.